### DISCUSSION TITLE
Add more time options to play from URL

### DISCUSSION THREAD

**Discussion by cybercuate** [AzuraCast/AzuraCast#7544] [Ideas]
![Captura de pantalla 2024-11-07 144352](https://github.com/user-attachments/assets/f23c190f-9cd1-4336-87a2-de347b23b4bc)

I have 1 mp3 file hosted on my http server that I want to play every "1 hour" or "after x number of songs" or "after x number of minutes". No, I can't just copy the file to the azuracast server because azuracast sometimes gives me the error "file cannot be processed".

Can you please add more options to play files hosted on URLs?. Scheduling on a fixed hour everyday is not "fun" for me

**Comment by BusterNeece** [comment]
I'm afraid I don't understand the question. What do you mean "more options to play files hosted on URLs?" What more options specifically would you be looking for?

Also, if you're having trouble processing a specific file, it may be worth filing that as a bug. There may be something wrong with the formatting of the file, but there also may be something wrong with _our_ processing of said file.

**Reply by gAlleb** [reply]
Hi Buster, I think he means 
>  I want to play every "1 hour" or "after x number of songs" or "after x number of minutes". 

options, that are provided for a normal Song-Based playlist.

<hr/>

> No, I can't just copy the file to the azuracast server because azuracast sometimes gives me the error "file cannot be processed".

@cybercuate, I think it might be easier to actually solve the "file problem" than adding functionality you are asking for. Maybe you could share the file that is causing errors? And maybe a bit more detailed log output when it happens: "file cannot be processed".

**Reply by cybercuate** [reply]
> Hi Buster, I think he means
> 
> > I want to play every "1 hour" or "after x number of songs" or "after x number of minutes".
> 
> options, that are provided for a normal Song-Based playlist.
> 
> > No, I can't just copy the file to the azuracast server because azuracast sometimes gives me the error "file cannot be processed".
> 
> @cybercuate, I think it might be easier to actually solve the "file problem" than adding functionality you are asking for. Maybe you could share the file that is causing errors? And maybe a bit more detailed log output when it happens: "file cannot be processed".

Sure, where can I send you my error logs? I'd like to keep private my station's info as much as possible.
Also, I noticed, if I try to upload the file with the web browser it uploads correctly. Errors start happening only with sftp and with a random number of songs. I tried to upload an album yesterday and some songs had the same error. Reuploading fixes them but that shouldn't be a good solution. Also. the refresh button doesn't work. I have to go back a directory and then come back to the same directory so I can see any changes on the files. I can send you screen captures

**Reply by gAlleb** [reply]
You can remove all the sensitive info before pasting, if you want. More can see it - higher the chance the problem will be resolved.


>  No, I can't just copy the file to the azuracast server because azuracast sometimes gives me the error "file cannot be processed".
>  Errors start happening only with sftp and with a random number of songs. I tried to upload an album yesterday and some songs had the same error.

So the issue is not with a certain file but happens randomly with random ones?
